webapp/src/shared/components/DispatchPage/DispatchPage.tsx

```diff
@@ -39,6 +75,11 @@ export default function DispatchPage<T>({
     return array.every((element) => elementChecker(element));
   };
 
+  const handleSearch = (value: string) => {
+    setSearch(value);
```

Comment:
I'm struggling to understand when do we need useCallback and when we don't. My guess so far, as a rule of thumb, is: if we don't need a dependency array, we don't need usecallback. Also, if we don't need to provide referential stability to the component we are passing the callback to, we don't need usecallback.
Here, I see we don't have nor need a dependency array, but we do pass it to SearchForm, that will keep reredering everytime the component renders, because the callback will be a different object every time.
So I'm not sure. Can you help with this?